### PR TITLE
todoman: disable failing test and unmark broken for darwin

### PR DESCRIPTION
**Pull request by johnjameswhitman**
###### Description of changes

The other day I updated my unstable channel and tried switching home-manager to a new generation on my x86 Macbook, but it failed. I tracked it down to `todoman`, which was disabled for darwin in [`37c633f`](https://github.com/NixOS/nixpkgs/commit/37c633f7ae518456af1f3064bd5c386e2de92c37#diff-d6f12ab586366800496d5e0658a7f2642b81342e9374f779b460df8860d46071R79) but seems to build fine for me. I also noticed #173671 (zero-hydra-failures PR) that mentions disabling some packages that were broken on hydra (the PR mentions `todoman`, but didn't actually disable it).

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin &larr; I'm wondering if [`37c633f`](https://github.com/NixOS/nixpkgs/commit/37c633f7ae518456af1f3064bd5c386e2de92c37#diff-d6f12ab586366800496d5e0658a7f2642b81342e9374f779b460df8860d46071R79) was addressing it failing on M1 specifically (cc @Mindavi)?
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Mindavi** [comment]
It was failing to build on hydra: https://hydra.nixos.org/job/nixpkgs/trunk/todoman.x86_64-darwin. Not sure about aarch64-darwin.

**Comment by johnjameswhitman** [comment]
The tests do take a while to run when I build it locally. I'll take a look at how to disable -- out of curiosity, do you prefer that over extending the timeout because of resource contention on hydra?

File: pkgs/applications/office/todoman/default.nix [hunk 1]
```diff
@@ -67,6 +67,7 @@ buildPythonApplication rec {
     "test_default_command"
     "test_main"
     "test_missing_cache_dir"
+    "test_sorting_fields"
```
Comment (johnjameswhitman) [inline pkgs/applications/office/todoman/default.nix:70]: Should we only do this if the tests are running on darwin?

Comment (Mindavi) [reply pkgs/applications/office/todoman/default.nix:70]: Yes please, it never fails on z86_64-linux AFAICT

**Comment by Mindavi** [comment]
Make sure to squash and observe the contributing guidelines regarding commit messages.

Something like this

```
pythonPackages.todoman: disable failing test and unmark broken for darwin
```

**Comment by Mindavi** [comment]
@ofborg build todoman

File: pkgs/applications/office/todoman/default.nix (hunk 1 above)
Comment (johnjameswhitman) [reply pkgs/applications/office/todoman/default.nix]: SG -- updated!

**Review by Mindavi — APPROVED** [review]
Thanks, this should be fine and it built now

**Pull request merged**